src/neuroconv/datainterfaces/ophys/basesegmentationextractorinterface.py

```diff
@@ -67,6 +67,8 @@ def run_conversion(
         stub_test: bool = False,
         stub_frames: int = 100,
         include_roi_centroids: bool = True,
+        iterator_options: Optional[dict] = None,
```

Comment:
What do you think on making this more specific by naming this `hdmf_iterator_options` and `hdmf_compression_options`?

Reply:
Too much typing, and no one needs to know that it has anything to do with HDMF underneath

Reply:
Also - just imagining - what if we one day have compression/iterators that aren't reliant on HDMF? I'd rather not have to add or modify the keyword argument name to correct that or worse, add a separate argument. Better to just have all that be a part of the dictionary

Reply:
Yes, I am on the fence on that one. I agree that that's probably leaking too much information to users which can be better described in the docstring. The concern comes from the fact that the word iterator has a wider meaning in Python that does not match with the specific and narrow meaning* of the word in HDMF and I wanted to indicate or disambiguate that. It does not come from a desire to let the users know that the we use THAT iterator.

What I mean is that the iterator here is a specific type of iterator that is focused mainly in solving the problem of writing data efficiently without loading everything into memory.  Maybe `iterative_write_options` would be a better description? That's what they use in their tutorial to describe the problem / context that the tool is solving:
https://pynwb.readthedocs.io/en/stable/tutorials/advanced_io/iterative_write.html#sphx-glr-tutorials-advanced-io-iterative-write-py